src/service.rs

```diff
@@ -116,28 +130,71 @@ impl Service {
     fn export_listener(&self) -> Result<Listener> {
         let mut filters = Vec::new();
 
-        // @TODO no way, move this to a function or something.
-        let mut buf = Vec::new();
-        prost::Message::encode(
-            &Router {
-                ..Default::default()
-            },
-            &mut buf,
-        )?;
+        fn encode(arg: impl prost::Message) -> Vec<u8> {
+            let mut buf = Vec::new();
+            prost::Message::encode(&arg, &mut buf).unwrap();
```

Comment:
Not important now, but you can use a `Result` here to return any errors and use `?`.  Same thing goes for most places where you use `unwrap()` in a fn than returns (or can return) a `Result<...>`.